### ISSUE TITLE
rule: `prefer-method-call`

### ISSUE THREAD

**Issue by paperclover** [DonIsaac/zlint#222]
## Description
Prefer method call syntax instead of passing the namespace's type as the first parameter.

## Examples

Examples of **incorrect** code for this rule
```zig
const A = struct {
    fn doSomething(a: A, arg: i32) void {};
    fn actOnPointer(a: *A, arg: i32) void {};
    comptime {
        var a: A = .{};
        doSomething(a, 4); // a.doSomething(4);
        actOnPointer(&a, 4); // a.actOnPointer(4);
        (&a).actOnPointer(2); // a.actOnPointer(4); // maybe separate rule
    }
};

```

Examples of **correct** code for this rule
```zig
const A = struct {
    fn doSomething(a: A, arg: i32) void {};
    comptime {
        var a: A = .{};
        a.doSomething(4);
        doSomething(.{}, 5); // opinion: transforming to method is uglier than direct.
        (A{ }).doSomething(4); // preserve this form if it was already used
    }
};

```
